newlib/libc/tinystdio/stdio_private.h

```diff
@@ -248,6 +257,14 @@ static inline void __flockfile_close(FILE *f) {
 #endif
 }
 
+#ifdef __clang__
```

Comment:
I think it would make sense to add a comment here that this exists to silence subtraction from NULL UBSan errors. Otherwise readers might be confused why this macro is used in some places but not others.

Reply:
Thanks, I've added a comment about this. The alternative was to make a conditional check for the pointer value, which couldn't be elided without checking for the sanitizer. Ick.